src/identity/IdentityProviderFactory.ts

```diff
@@ -0,0 +1,77 @@
+import type { AnyObject,
+  CanBePromise,
+  interactionPolicy as InteractionPolicy,
+  KoaContextWithOIDC,
+  Configuration,
+  Account } from 'oidc-provider';
+import { Provider } from 'oidc-provider';
+
+import type { IdPConfigurationFactory } from './configuration/IdPConfigurationFactory';
+
+export class IdentityProviderFactory {
```

Comment:
Needs comments as to what it does and why.

Reply:
Done

Reply:
Yes, but how come we need a factory as opposed to just a class that we instantiate?

Reply:
Changed name IdpConfigurationFactory to "IdpConfigurationGenerator"